tests/test_block.py

```diff
@@ -31,3 +34,57 @@ def test_save_with_no_instance(self):
             self.assertTrue('Block is not associated with an instance' in
                             context.exception.args[0])
             self.assertFalse(b._put.called)
+
+        def test_load_template(self):
+            from .example_data import SimulatorFastTemplate
+            instance = lambda: None
+            instance.droplog = print
+            btype = 'SimulatorFast'
+            b = Block('sim', btype)
+            b.load_template(btype, SimulatorFastTemplate, instance)
+            from pprint import pprint
+            print()
+            pprint(b.template)
+            self.assertRaises(TypeError, setattr, b.template, 'type', 'foo')
+
+        def test_typed_config(self):
+            from .example_data import SimulatorFastTemplate, SimulatorFastConfig
+            instance = lambda: None
+            instance.droplog = print
+            btype = 'SimulatorFast'
+            b = Block('sim', btype, config=SimulatorFastConfig)
+            b.load_template(btype, SimulatorFastTemplate, instance)
+            b.config.interval.days = 10
+            self.assertEqual(b.config.interval.days, 10)
+            self.assertRaises(ValueError, setattr,
+                              b.config.interval, 'days', 'bad')
+
+        def test_load_all_templates(self):
+            from .example_data import BlocksTemplatesAll
+            instance = lambda: None
+            instance.droplog = MagicMock()
+            for btype, template in BlocksTemplatesAll.items():
+                b = Block(btype, btype)
+                b.load_template(btype, template, instance)
+
+        def test_load_all_configs(self):
+            from .example_data import BlocksTemplatesAll, BlocksConfigsAll
+            instance = lambda: None
+            instance.droplog = MagicMock()
+
+            blocks_types = {}
+            for btype, template in BlocksTemplatesAll.items():
+                b = Block(btype, btype)
+                b.load_template(btype, template, instance)
+                blocks_types[btype] = b
+
+            blocks = {}
+            for bname, config in BlocksConfigsAll.items():
+                # import ipdb; ipdb.set_trace()
+                btype = config['type']
+                b = deepcopy(blocks_types[btype])
+                b.config = config
+                blocks[bname] = b
+
+            droplog = instance.droplog
+            self.assertEqual(droplog.call_count, 2)
```

Comment:
The main thing it is doing is verifying that it can load a large range of configs. It's kind of a "throw everything I can at the test and make sure it works" kinda test. Remember that the TypedDict is verifying types throughout this whole process, so the fact that it can load all those configs is significant. 

The drop rate is just an artifact of all the loads.